cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -183,30 +183,59 @@ func removeIncompleteShards(dir string) {
 	}
 }
 
-func removeAll(shards []shard) {
+func removeAll(shards ...shard) {
+	// Note on error handling here: We only expect this to fail due to
+	// IsNotExist, which is fine. Additionally this shouldn't fail
+	// partially. But if it does, and the file still exists, then we have the
+	// potential for a partial index for a repo. However, this should be
+	// exceedingly rare due to it being a mix of partial failure on something in
+	// trash + an admin re-adding a repository.
```

Comment:
Worth capturing these errors in a Prometheus metric and put it on the Zoekt dashboard?